File: src/main/java/eschool/courses/model/Section.java
package eschool.courses.model;

import java.util.Set;

import javax.jdo.annotations.*;

@PersistenceCapable
public class Section {
	@PrimaryKey
	@Persistent(valueStrategy=IdGeneratorStrategy.INCREMENT)	
	private long id;
	private Course course;
	@Unique
	private String sectionId;
	private Set<Term> terms;
	private Set<Period> periods;
	private Room room;
	
	public Section() {}
	
	public Section(Course course, String sectionId, Set<Term> terms, Set<Period> periods, Room room) {
		this.course = course;
		this.sectionId = sectionId;
		this.terms = terms;
		this.periods = periods;
		this.room = room;
	}

	public Course getCourse() {
		return course;
	}

	public void setCourse(Course course) {
		this.course = course;
	}

	public String getSectionId() {
		return sectionId;
	}

	public void setSectionId(String sectionId) {
		this.sectionId = sectionId;
	}

	public Set<Term> getTerms() {
		return terms;
	}

	public void setTerms(Set<Term> terms) {
		this.terms = terms;
	}

	public Set<Period> getPeriods() {
		return periods;
	}

	public void setPeriods(Set<Period> periods) {
		this.periods = periods;
	}

	public Room getRoom() {
		return room;
	}

	public void setRoom(Room room) {
		this.room = room;
	}

	public long getId() {
		return id;
	}
}

File: src/main/scala/eschool/courses/model/Section.scala
package eschool.courses.model

import scala.collection.JavaConverters._

import jdo.Id

import javax.jdo.annotations._

class Section extends Id[Long] {
  private[this] var _course: Course = _
  @Unique
  private[this] var _sectionId: String = _
  @Element(types=Array(classOf[Term]))
  @Join
  private[this] var _terms: java.util.Set[Term] = _
  @Element(types=Array(classOf[Period]))
  @Join
  private[this] var _periods: java.util.Set[Period] = _
  private[this] var _room: Room = _
  
  def this(course: Course, sectionId: String, terms: Set[Term], periods: Set[Period], room: Room) = {
    this()
    _course = course
    _sectionId = sectionId
    terms_=(terms)
    periods_=(periods)
    _room = room
  }
  
  def course: Course = _course
  def course_=(theCourse: Course) { _course = theCourse }
  
  def sectionId: String = _sectionId
  def sectionId_=(theSectionId: String) { _sectionId = theSectionId }
  
  def terms: Set[Term] = _terms.asScala.toSet
  def terms_=(theTerms: Set[Term]) { _terms = theTerms.asJava }
  
  def room: Room = _room
  def room_=(theRoom: Room) { _room = theRoom }
  
  def periods: Set[Period] = _periods.asScala.toSet
  def periods_=(thePeriods: Set[Period]) { _periods = thePeriods.asJava }
  
  def periodNames: String = {
    periods.map(_.name).mkString(", ")
  }
}
